### DISCUSSION TITLE
Dragging files in the view

### DISCUSSION THREAD

**Discussion by jpcima** [cycfi/elements#335] [Elements]
Hello, I wish to develop the ability to drop files into an area of the view.

The file will be dragged by mouse, the movement should be tracked by the UI, and it could react to the file being hovered on top of various elements.

For instance, I would like an event that allows to modify the cursor as the file is dragged, for example :heavy_plus_sign: or :no_entry_sign:, and then to take action when the drop gesture finishes.

The use case is to drag-drop a sound file into a musical program for instance.

**Comment by Xeverous** [comment]
First I would like the ability to have a "hower" element that is attached to the cursor - this will open space for fancy tooltips and to hide the cursor and display such element instead. I'm not sure if all the machinery required to do such mouse-following element (rendered on top of everything else) is present.

Then, implementing actual drag & drop - this is OS specific as it requires integration with OS cross-window/cross-program events.

Said differently, this issue will take some time.
